services/report/__init__.py

```diff
@@ -1173,6 +1191,74 @@ def save_report(self, commit: Commit, report: Report, report_code=None):
         )
         return {"url": url}
 
+    def _determine_cffs_and_depths_in_db(
+        self, db_session, commit: Commit
+    ) -> Optional[Dict[int, int]]:
+        depths_with_cff_upload_ids = defaultdict(list)
+
+        existing_cff_uploads: Optional[List[Upload]] = (
+            db_session.query(Upload)
+            .filter_by(
+                report_id=commit.report.id_,
+                upload_type=SessionType.carriedforward.value,
+            )
+            .all()
+        )
+
+        if existing_cff_uploads:
+            for upload in existing_cff_uploads:
+                parent_depth = upload.upload_extras.get("parent_depth")
+                if parent_depth:
+                    depths_with_cff_upload_ids[parent_depth].append(upload.id)
+
+            log.info(
+                "Existing cffs and their respective depths",
+                extra=dict(
+                    commit=commit.commitid,
+                    repo=commit.repoid,
+                    existing_uploads=existing_cff_uploads,
+                ),
+            )
+
+        return depths_with_cff_upload_ids
+
+    def _possibly_delete_existing_cffs(
+        self,
+        db_session,
+        commit: Commit,
+        carryforward_report: Report,
+    ):
+        cffs_and_depths_in_db = self._determine_cffs_and_depths_in_db(
+            db_session=db_session, commit=commit
+        )
+
+        if not cffs_and_depths_in_db:
+            return
+
+        # Gets first 'parent_depth' it finds, as all cffs should be carried forward
```

Comment:
Shouldn't it get the minimum 'parent_depth' instead of the first?

Reply:
The next four lines do that, it goes through the Report object which contains the uploads that will be carryforwarded and will determine the depth they have. I'm initializing `carryforward_report_depth` with the arbitrary depth of 1 and in case line 1241 doesn't run, so the variable has a non-null value. I update it shortly after in line 1243 if a depth for the carryforward reports are found, does that make sense?

Reply:
well... no. The question is if the code should get the _smallest_ 'parent_depth' from all the sessions in the report. Both the comment and the code are getting the 1st parent_depth that appears. Those values can be different depending on the order of sessions that were carried forward, and you made no claims about the order of the sessions

asked differently: Is the 1st session that appears always the smallest one?